test/functional/test_framework/wallet.py

```diff
@@ -55,25 +80,41 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
+    def get_utxos(self, amount):
+        """Removes and returns the last {amount} utxos from the wallet"""
+        assert amount <= len(self._utxos)
+        ret = self._utxos[-amount:]
+        self._utxos = self._utxos[:-amount]
+        return ret
+
+    def send_self_transfer(self, *, fee_rate=Decimal("0.00003"), from_node, utxo_to_spend=None, send_values=None, nSequence=0, nVersion=1):
         """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
         self._utxos = sorted(self._utxos, key=lambda k: k['value'])
         utxo_to_spend = utxo_to_spend or self._utxos.pop()  # Pick the largest utxo (if none provided) and hope it covers the fee
-        vsize = Decimal(96)
-        send_value = satoshi_round(utxo_to_spend['value'] - fee_rate * (vsize / 1000))
-        fee = utxo_to_spend['value'] - send_value
-        assert send_value > 0
-
+        send_values = send_values or [utxo_to_spend['value']]
+        assert sum(send_values) <= utxo_to_spend['value']
+        vsize = Decimal(53 + 43 * len(send_values))
         tx = CTransaction()
-        tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']))]
-        tx.vout = [CTxOut(int(send_value * COIN), self._scriptPubKey)]
+        tx.nVersion = nVersion
+        tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']), nSequence=nSequence)]
+
+        # Distribute the fee evenly among all outputs
+        fee_per_output = fee_rate * vsize / 1000 / len(send_values)
+
+        total_sent = 0
+        for send_value in send_values:
+            output_value = satoshi_round(send_value - fee_per_output)
+            total_sent += output_value
+            tx.vout.append(CTxOut(int(output_value * COIN), self._scriptPubKey))
+
         tx.wit.vtxinwit = [CTxInWitness()]
         tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
         tx_hex = tx.serialize().hex()
 
         tx_info = from_node.testmempoolaccept([tx_hex])[0]
-        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
-        from_node.sendrawtransaction(tx_hex)
+        for i, send_value in enumerate(send_values):
+            self._utxos.append({'txid': tx_info['txid'], 'vout': i, 'value': satoshi_round(send_value - fee_per_output)})
+
+        assert_equal(tx_info['txid'], from_node.sendrawtransaction(tx_hex))
         assert_equal(tx_info['vsize'], vsize)
-        assert_equal(tx_info['fees']['base'], fee)
```

Comment:
I believe for the fees assertion, you weren't taking the whole fees of the tx into consideration.  If you print(tx) in line 109, you'll see tx with one input and lots of outputs. `fee_per_output` only calculated tx for a single set of output, but here you need to calculate fee for the whole tx. I tried doing something like this and the assertion worked! 

```
...
total_base_fee = 0
        for send_value in send_values:
            output_value = satoshi_round(send_value - fee_per_output)
            total_base_fee += satoshi_round(send_value - output_value)
            total_sent += output_value
            tx.vout.append(CTxOut(int(output_value * COIN), self._scriptPubKey))
...
assert_equal(tx_info['fees']['base'], total_base_fee)
```

Reply:
Thanks, that was helpful - I actually had something similar before but I think my mistake was forgetting to use `satoshi_round` on `fee_per_output`, and then I just convinced myself that I didn't know what fees-base actually represented.